includes/field_truncation.inc

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * @file
+ * Class to assist in field truncation, paying special attention to HTML.
+ */
+
+/**
+ * Islandora Field Truncation.
+ */
+class IslandoraSolrFieldTruncation {
+  protected $writer;
+  protected $output = '';
+  protected $wordSafe;
+  protected $addEllipsis;
+  protected $wordsafeLength;
+
+  const GENERAL_DOM = <<<EOXML
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+</head>
+<body>@replace@</body>
+</html>
+EOXML;
+
+  const XHTML_URI = 'http://www.w3.org/1999/xhtml';
+
+  /**
+   * Class Constructor function.
+   *
+   * @param string $text
+   *   The text to truncate.
+   * @param int $desired_length
+   *   The desired length to truncate to.
+   * @param bool $word_safe
+   *   Attempt to truncate on a word boundary.
+   * @param bool $add_ellipsis
+   *   Add '...' to the end of the truncated string.
+   * @param int $wordsafe_length
+   *   If $word_safe is true, the minimum acceptable length for truncation.
+   *   No effect if $word_safe is false.
+   */
+  public function __construct($text, $desired_length, $word_safe = TRUE, $add_ellipsis = TRUE, $wordsafe_length = 1) {
+    $this->wordSafe = $word_safe;
+    $this->addEllipsis = $add_ellipsis;
+    $this->wordsafeLength = $wordsafe_length;
+
+    if (strpos($text, '<') === FALSE && strpos($text, '>') === FALSE) {
```

Comment:
I am a little hesitant to implement a reg ex here, feels like there may be more of a margin of error. As well, according to the [preg_match()](http://php.net/manual/en/function.preg-match.php) page it looks like it would be faster to use a strpos().